mypy/docstrings.py

```diff
@@ -0,0 +1,27 @@
+from typing import List, Tuple, Dict
+from mypy.types import Type, CallableType, AnyType
+from mypy.nodes import Argument
+
+
+def parse_docstring(docstring: str, line: int) -> Tuple[Dict[str, Type], Type]:
+    """
+    Parse a docstring and return type representations.  This function can
+    be overridden by third-party tools which aim to add typing via docstrings.
+
+    Returns a 2-tuple: dictionary of arg name to Type, and return Type.
+    """
+    return None, None
+
+
+def make_callable(args: List[Argument], type_map: Dict[str, Type],
+                  ret_type: Type) -> CallableType:
+    if type_map is not None:
```

Comment:
Why is this function in this file? It seems to be used only by parse.py so it really belongs there, unless you also plan to monkey-patch this.

There's a missing Optional in the signature. But it almost feels like the caller should check whether type_map is None and not even call this.

(Also, why is this only needed by parser.py and not by fastparse*.py?)